### PR TITLE
Add documentation

### PR DESCRIPTION
**Pull request by sol**

### CODE REVIEW

File: src/Test/Mockery/Directory.hs [hunk 1]
```diff
@@ -7,12 +7,19 @@ import           Control.Exception
 import           System.Directory
 import           System.IO.Temp (withSystemTempDirectory)
 
+-- |
+-- Run given action with the current working directory set to a temporary
+-- directory.  The temporary directory in created before the action is run and
```
Comment (soenkehahn) [inline src/Test/Mockery/Directory.hs]: `in` -> `is`.

File: src/Test/Mockery/Directory.hs [hunk 2]
```diff
@@ -7,12 +7,19 @@ import           Control.Exception
 import           System.Directory
 import           System.IO.Temp (withSystemTempDirectory)
 
+-- |
+-- Run given action with the current working directory set to a temporary
```
Comment (jkarni) [inline src/Test/Mockery/Directory.hs:11]: "Run _the_ given action"?

File: src/Test/Mockery/Directory.hs (hunk 1 above)
Comment (soenkehahn) [inline src/Test/Mockery/Directory.hs]: Maybe we should add a comment about the fact that the working directory is restored?

Comment (jkarni) [inline src/Test/Mockery/Directory.hs]: "_is_ created"

**Comment by soenkehahn** [comment]
LGTM.

File: src/Test/Mockery/Directory.hs (hunk 1 above)
Comment (sol) [inline src/Test/Mockery/Directory.hs]: thx

File: src/Test/Mockery/Directory.hs [hunk 3]
```diff
@@ -7,12 +7,22 @@ import           Control.Exception
...
+-- directory.
+--
+-- The directory is created before the action is run.  After the the action has
+-- completed the diretory is removed and the current working diretory is
```
Comment (soenkehahn) [inline src/Test/Mockery/Directory.hs]: `diretory` -> `directory`. Other than that LGTM.

File: src/Test/Mockery/Directory.hs [hunk 4]
```diff
@@ -7,12 +7,22 @@ import           Control.Exception
...
+-- restored to its original value.
 inTempDirectory :: IO a -> IO a
 inTempDirectory action = withSystemTempDirectory "hspec" $ \path -> do
   bracket getCurrentDirectory setCurrentDirectory $ \_ -> do
     setCurrentDirectory path
     action
 
+-- |
+-- Similar to `inTempDirectory`, but the temporary diretory will have the
```
Comment (soenkehahn) [inline src/Test/Mockery/Directory.hs]: Same here.

**Pull request merged**